observability/observability.go

```diff
@@ -96,6 +134,18 @@ func ContextWithReceiverName(ctx context.Context, receiverName string) context.C
 	return ctx
 }
 
+// RecordIngestionBlockedMetrics records metrics related to the receiver responses
+// when the host blocks ingestion. If back pressure is disabled the metric for
+// respective data loss is recorded.
+// Use it with a context.Context generated using ContextWithReceiverName().
+func RecordIngestionBlockedMetrics(ctxWithTraceReceiverName context.Context, backPressureDisabled bool) {
```

Comment:
If you keep the BackPressureState type I recommend using it here too instead of bool.